contrib/gin-gonic/gin/gintrace_test.go

```diff
@@ -340,6 +341,54 @@ func TestResourceNamerSettings(t *testing.T) {
 	})
 }
 
+func TestIgnoreRequestSettings(t *testing.T) {
+	tests := []struct {
+		url       string
+		spanCount int
+	}{
+		{
+			url:       "/skip",
+			spanCount: 0,
+		},
+		{
+			url:       "/skipfoo",
+			spanCount: 0,
+		},
+		{
+			url:       "/OK",
+			spanCount: 1,
+		},
+	}
+
+	router := gin.New()
+	router.Use(Middleware("foobar", WithIgnoreRequest(func(c *gin.Context) bool {
+		return strings.HasPrefix(c.Request.URL.Path, "/skip")
+	})))
+
+	router.GET("/OK", func(c *gin.Context) {
+		c.Writer.Write([]byte("OK"))
+	})
+
+	router.GET("/skip", func(c *gin.Context) {
+		c.Writer.Write([]byte("Skip"))
+	})
+
+	for _, test := range tests {
+		t.Run(test.url, func(t *testing.T) {
+			mt := mocktracer.Start()
+			defer mt.Stop()
+
+			r := httptest.NewRequest("GET", "http://localhost"+test.url, nil)
+			w := httptest.NewRecorder()
+
+			router.ServeHTTP(w, r)
+
+			spans := mt.FinishedSpans()
+			assert.Equal(t, test.spanCount, len(spans))
```

Comment:
```suggestion
			mt := mocktracer.Start()
			defer mt.Reset()

			r := httptest.NewRequest("GET", "http://localhost"+test.url, nil)
			router.ServeHTTP(httptest.NewRecorder(), r)
			assert.Equal(t, test.spanCount, len(mt.FinishedSpans()))
```
No need to create variables that are not used. Also, we can use `Reset` instead of `Stop`, on the mocktracer.

Reply:
Sounds good, addressed in latest commit.